### ISSUE TITLE
[BUG]  Unable to Separate Workflow Definitions and Instances in Different Databases with Elsa 3

### ISSUE THREAD

**Issue by MohsenAra** [elsa-workflows/elsa-core#6027]
Description:
I'm using Elsa 3 and trying to store workflow definitions and workflow instances in separate SQL Server databases. However, regardless of the separate configuration for both, only the connection string specified for workflow instances is being used for both workflows (instances and definitions). I suspect the issue is related to using a single DbContext for both workflow definitions and workflow instances.

Current Configuration:
I am configuring Elsa using the following connection strings:

`var instanceConnectionString = "Data Source=.;Initial Catalog=elsa_instances;User ID=sa;Password=123;multipleactiveresultsets=True;Enlist=false;Application Name=Instances;Encrypt=false;";
var definitaionConnectionString = "Data Source=.;Initial Catalog=elsa_definitions;User ID=sa;Password=123;multipleactiveresultsets=True;Enlist=false;Application Name=Definitions;Encrypt=false;";

`

**Comment by sfmskywalker** [comment]
You're right, the intent is to allow the application to separate storage between workflow definitions and workflow instances. Clearly, this doesn't work when using the same DbContext. The DB context factory seems to be bound to the specified type of DbContext. I think that the most practical solution might indeed be to separate the ManagementDbContext into WorkflowDefinitions and WorkflowInstances.
